### PR TITLE
Mongodb panache/mutiny

### PR DESCRIPTION
**Pull request by loicmathieu**
@cescoffier I wanted to play with Mutiny so I integrate it in MongoDB with Panache instead of Axle.

It works !

I have one big question, for the moment it uses Mutiny on the inside but it still exposes CompletionStage/Publisher.

Is the plan to expose Mutiny at the API level or only used it on the inside and expose standard reactive types at the API level ?

### CODE REVIEW

**Comment by cescoffier** [comment]
Damned, didn't see that, I've opened another PR.

**Comment by cescoffier** [comment]
Ok, we would need to merge both, because you did something I didn't - implement the previous API with the new one.

**Comment by loicmathieu** [comment]
@cescoffier I didn't implement the previous API with the new one, I replace the previous API with Mutiny.
As I said in the Mutiny epic, the Axle implementation has been done recently for 1.3 so not already used. That's why we can replace it. No need to deprecate it.

The main difference between your PR and mine is that you return Mutiny types where I stick to CompletionStage and Publisher. This was a question I ask on the description of my PR and now it is answered ;)

So two solution here : update my PR to Mutiny return types and merge it or update you PR by emoving the Axle implementation and merge it. WDYT ?

**Comment by loicmathieu** [comment]
@cescoffier updated to return the Mutiny types at the API level as we discussed.

**Comment by loicmathieu** [comment]
@FroMage I updated the reactive MongoDB with Panache implementation to use Mutiny. 
If you want to have a look before it's merged I think that @cescoffier would want it merged quickly to meet the 1.3 target ...

**Comment by loicmathieu** [comment]
I had to add the `quarkus-smallrye-context-propagation` inside the integration test to make it works. 

I don't really understand why and I think that it should be integrated by resteasy or panache because a user cannot know that it must include it ...

**Comment by cescoffier** [comment]
The context-propagation issue is due to a weird chain of dependency. I will investigate that next.

File: docs/src/main/asciidoc/mongodb-panache.adoc
Comment (cescoffier) [inline docs/src/main/asciidoc/mongodb-panache.adoc]: They should return a `Multi` (which implement `Publisher`)

**Review by cescoffier — CHANGES_REQUESTED** [review]
a very small comment in the doc where it should mention `Multi` instead of `Publisher`.

File: docs/src/main/asciidoc/mongodb-panache.adoc
Comment (loicmathieu) [reply docs/src/main/asciidoc/mongodb-panache.adoc]: updated

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
Thanks @loicmathieu - Waiting for CI!

**Comment by machi1990** [comment]
Merged thanks @loicmathieu @cescoffier

**Pull request merged**
